packages/shorebird_code_push_api/lib/src/middleware/http_client_provider.dart

```diff
@@ -1,12 +1,9 @@
 import 'package:http/http.dart' as http;
 import 'package:shelf/shelf.dart';
-import 'package:shorebird_code_push_api/src/http_client.dart';
 import 'package:shorebird_code_push_api/src/provider.dart';
 
-Middleware httpClientProvider(String key) {
-  return provider<Future<http.Client>>(
-    (_) async => _httpClient ??= createClient(key),
-  );
-}
+http.Client? _httpClient;
 
-Future<http.Client>? _httpClient;
+Middleware httpClientProvider() {
```

Comment:
Was the async removal automatic?  Feels a bit odd to leave the provider but remove the async.